specutils/io/registers.py

```diff
@@ -48,19 +48,29 @@ def wrapper(*args, **kwargs):
                 return False
         return wrapper
 
-    identifier = identifier_wrapper(identifier)
-
     def decorator(func):
         io_registry.register_reader(label, dtype, func)
 
-        # If the identifier is not defined, but the extensions are, create
-        # a simple identifier based off file extension.
-        if identifier is None and extensions is not None:
-            io_registry.register_identifier(
-                label, dtype, lambda *args, **kwargs: any([args[0].endswith(x)
-                                                           for x in extensions]))
+        if identifier is None:
+            # If the identifier is not defined, but the extensions are, create
+            # a simple identifier based off file extension.
+            if extensions is not None:
+                logging.info("'{}' data loader provided for {} without "
+                             "explicit identifier. Creating identifier using "
+                             "list of compatible extensions".format(
+                                 label, dtype.__name__))
+                id_func = lambda *args, **kwargs: any([args[0].endswith(x)
+                                                          for x in extensions])
+            # Otherwise, create a dummy identifier
+            else:
+                logging.warn("'{}' data loader provided for {} without "
+                             "explicit identifier or list of compatible "
+                             "extensions".format(label, dtype.__name__))
+                id_func = lambda *args, **kwargs: False
```

Comment:
I could honestly see this just as easily being `True`, in which case it will blindly identify *any* file as compatible. I'm not entirely thrilled with either option.

A third way would be to simply raise an error in the case that neither an identifier nor a list of compatible extensions was provided. This would mean custom identifiers would be required to provide one or the other, though.

Reply:
Currently, the specviz loader wizard does not create identifier functions. This means that, right now, specviz cannot use any custom loader created through the loader wizard. I'm inclined to say that if no explicit identifier function is given, and no extension information is provided, blindly accept the input, and catch it later when/if it fails.